mmedit/models/common/ensemble.py

```diff
@@ -0,0 +1,71 @@
+# Copyright (c) OpenMMLab. All rights reserved.
+import torch
+import torch.nn as nn
+
+
+class SpatialTemporalEnsemble(nn.Module):
+    """ Apply spatial and temporal ensemble and compute outputs
+
+    Args:
+        is_temporal_ensemble (bool, optional): Whether to apply ensemble
+            temporally. If True, the sequence will also be flipped temporally.
+            If the input is an image, this argument must be set to False.
+            Default: False.
+
+    Returns:
+        Tensor: Model outputs given the ensembles.
+    """
+
+    def __init__(self, is_temporal_ensemble=False):
+
+        super().__init__()
+
+        self.is_temporal_ensemble = is_temporal_ensemble
+
+    def _transform(self, imgs, mode):
+        is_single_image = False
+        if imgs.ndim == 4:
+            if self.is_temporal_ensemble:
+                raise ValueError('"is_temporal_ensemble" must be False if '
+                                 'the input is an image.')
+            is_single_image = True
+            imgs = imgs.unsqueeze(1)
+
+        if mode == 'vertical':
+            imgs = imgs.flip(4).clone()
+        elif mode == 'horizontal':
+            imgs = imgs.flip(3).clone()
+        elif mode == 'transpose':
+            imgs = imgs.permute(0, 1, 2, 4, 3).clone()
+
+        if is_single_image:
+            imgs = imgs.squeeze(1)
```

Comment:
I think squeeze() should be placed before flip() and permute()

Reply:
`unsqueeze` is to make sure the size of the tensor is of `(n, t, c, h, w)`. Therefore the operations do not need to distinguish `image` and `video` inputs.

After that it is squeezed.